types/src/era_id.rs

```diff
@@ -64,6 +64,11 @@ impl EraId {
         EraId::from(self.0 + 1)
     }
 
+    /// Returns the current era plus `x`, or `None` if that would overflow
+    pub fn checked_add(&self, x: u64) -> Option<EraId> {
```

Comment:
Since `U512` got `CheckedOp` traits implemented, I wouldn't mind if we'd do it here as well if we're at it (same for existing `checked_sub,` but that was likely my fault for not introducing it in the first place)

Reply:
Adding `CheckedOp` for `EraId` requires more an `Add<EraId` impl, and Im not too keen on that cause we can rely on the underlying `u64` Checked math to ensure we don't overflow, and ensure that we only can add `u64` to `EraId`